common/dbus/platform/asio_libdbus/dbus.cpp

```diff
@@ -578,6 +578,178 @@ optional::optional<DBusMethodHandler<ExpectedStringPair>> DBusObject::GetMethodH
 	}
 }
 
+error::Error DBusServer::InitializeConnection() {
+	auto err = DBusPeer::InitializeConnection();
+	if (err != error::NoError) {
+		return err;
+	}
+
+	DBusError dbus_error;
+	dbus_error_init(&dbus_error);
+
+	// We could also do DBUS_NAME_FLAG_ALLOW_REPLACEMENT for cases where two of
+	// processes request the same name, but it would require handling of the
+	// NameLost signal and triggering termination.
+	if (dbus_bus_request_name(
+			dbus_conn_.get(), service_name_.c_str(), DBUS_NAME_FLAG_DO_NOT_QUEUE, &dbus_error)
+		== -1) {
+		dbus_conn_.reset();
+		auto err = MakeError(
```

Comment:
Nit: err is already defined.

Reply:
yeah, but not it this scope and it's here only to split the code into multiple lines, really.